test/session.js

```diff
@@ -1665,6 +1694,40 @@ describe('session()', function(){
         })
       })
 
+      it('should error when the session is corrupt', function (done) {
+        var store = new session.MemoryStore()
+        var server = createServer({ store: store }, function (req, res) {
+          if (req.url === '/') {
+            req.session.active = true
+            res.end('session created')
+            return
+          }
+
+          store.clear(function (err) {
+            if (err) return done(err)
+            
+            store.get = function returnCorruptSession(sid, callback) {
```

Comment:
`store.get` needs to be overwritten here instead of on line 1719 because otherwise the error is caught (and surfaced gracefully) in the wrong spot and `.reload()` never executes.